core/commands/checkout.py

```diff
@@ -53,10 +54,17 @@ class GsCheckoutNewBranchCommand(WindowCommand, GitCommand):
     def run(self):
         sublime.set_timeout_async(self.run_async)
 
-    def run_async(self):
-        self.window.show_input_panel(NEW_BRANCH_PROMPT, "", self.on_done, None, None)
+    def run_async(self, name=""):
+        self.window.show_input_panel(NEW_BRANCH_PROMPT, name, self.on_done, None, None)
 
     def on_done(self, branch_name):
+        pattern = r"^(?!\.|.*\.\..*|.*@.*|\/)[a-zA-Z0-9\-\_\/\.]+(?<!\.lock)(?<!\/)(?<!\.)\b$"
+        match = re.match(pattern, branch_name)
+        if not match:
+            sublime.status_message("`{}` is a invalid branch name.".format(branch_name))
```

Comment:
Please change this to an error modal message - the status message bar can be easy to miss.